packages/ui-react/src/components/Payment/Payment.tsx

```diff
@@ -219,23 +221,34 @@ const Payment = ({
                   </Link>
                 </p>
               ) : (
-                showChangeSubscriptionButton && (
-                  <Button
-                    className={styles.upgradeSubscription}
-                    label={t('user:payment.change_subscription')}
-                    disabled={!canRenewSubscription && activeSubscription.status === 'cancelled'}
-                    onClick={() => {
-                      if (offers.length > 1 && !canRenewSubscription) {
-                        setIsChangingOffer(true);
-                      } else {
-                        onUpgradeSubscriptionClick?.();
-                      }
-                    }}
-                    fullWidth={isMobile}
-                    color="primary"
-                    data-testid="change-subscription-button"
-                  />
-                )
+                <>
+                  {canSwitchSubscription && showChangeSubscriptionButton && (
```

Comment:
I am hesitant to add additional complexity here introducing a new condition + new "cancel" button (third one?)

Reply:
Yes i will close this pr.